### ISSUE TITLE
What is the difference between flexible array and varying array?

### ISSUE THREAD

**Issue by ujos** [burlachenkok/CPP_from_1998_to_2020#9]
I'm not sure I understand the text below:

> 9. Support of varying array array. In C99, the last member of a structure can be a flexible array. In C++ it's possible as well.
> 10. In C99, but not in C++, there is a flexible arrays. That arrays with a size specified via a non-const variable. In C99, to pass such arguments to a function, use the notation void g(int pp[*], int k){}.

What is the difference between `flexible array` and `varying array`?

In C++ there is _open array_:

```cpp
void foo(int arr[], size_t arrSize)
{
}
```

From other side `variable array` in the documentation is described as `flexible array` here: https://wiki.sei.cmu.edu/confluence/display/c/DCL38-C.+Use+the+correct+syntax+when+declaring+a+flexible+array+member

**Comment by burlachenkok** [comment]
I have added C99 standard and description what is `flexible array`. It's a popular thing used pretty often in any kernel code for Windows and Linux.

Thanks for reference, such as:
https://wiki.sei.cmu.edu/confluence/display/c/DCL38-C.+Use+the+correct+syntax+when+declaring+a+flexible+array+member

But I try to use standards, when it's enough.

**Comment by ujos** [comment]
Ok, now it is more clear, however still I do not understand what is purpose of the `[*]`. MSVC does not support it. GCC does support but in the function prototype only. What is the difference between these below:
```c
void foo(int arr[], size_t arrSize);
void bar(int arr[*], size_t arrSize);
```

**Comment by ujos** [comment]
>If the size is not present, the array type is an incomplete type. If the size is * instead of
being an expression, the array type is a variable length array type of unspecified size,
which can only be used in declarations with function prototype scope;122) such arrays are
nonetheless complete types.

**Comment by burlachenkok** [comment]
```
> void foo(int arr[], size_t arrSize);
> void bar(int arr[*], size_t arrSize);
> ```

- I have added incompleted type information into the glossary
- I have updated the variable-length array description
- I have removed a variable-length array that is confusing [*] notation.

> What is the difference between these below?

I don't know. I thought construction with `[*]` allows to have an incomplete type, but it's not the case.
(In experiments with `gcc -x c --std=c99` through https://godbolt.org/)
